lib/clients/BaseClient.py

```diff
@@ -531,6 +531,48 @@ def decrypt_and_extract_tarball_of_directory(self, encrypted_tarball_name, direc
             return result
         return None
 
+    def create_tarball_of_directory(self, directory_to_tar, tarball_name):
+        """Create a tarball of a directory.
+
+        :param directory_to_tar: the path to the directory to be archived
+        :param tarball_name: the path where to store the resulting archive
+
+        :Example:
+            ::
+                tar cvzpf ACS-Demo.tar.gz -C ACS-Demo .
+                iaas_client.create_tarball_of_directory('/tmp/backup/files.tar.gz', '/var/vcap/store/blueprint/files')
+        """
+        self.logger.info(
+            '[COMPRESSION] Started creating and copying a tarball ...')
+        result = self.shell('tar cvzpf {} -C {} .'
+                            .format(tarball_name, directory_to_tar), False)
+        self.logger.info('[ENCRYPTION] ... finished.')
+        return result
+
+    def extract_tarball_of_directory(self, tarball_name, directory_to_extract):
+        """Extract an encrypted tarball of a directory.
+
+        :param arball_name: the path to the directory to be extracted
```

Comment:
typo "arball_name"

Reply:
`the path to the directory to be extracted` --> Is it a directory or file?